packages/builder/builder-webpack-provider/src/webpackPlugins/CheckSyntaxPlugin/helpers/utils.ts

```diff
@@ -0,0 +1,14 @@
+import { CheckSyntaxExclude } from './type';
+
+export function checkIsExcludeSource(
+  path: string,
+  exclude?: CheckSyntaxExclude,
+) {
+  if (!exclude) {
+    return false;
+  }
+
+  const excludes = Array.isArray(exclude) ? exclude : [exclude];
+
+  return excludes.some(reg => new RegExp(reg).test(path));
```

Comment:
This behavior is different from the `exclude` of webpack rules, we have two options:

1. Only support `RegExp | RegExp[]`
2. Make the behavior consistent with webpack exclude.

Reply:
Done~